### PR TITLE
Add MusicOnHoldEvent

### PR DESCRIPTION
**Pull request by moldypenguins**
Adds MusicOnHoldEvent.

### CODE REVIEW

File: Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			/// <summary>
+			/// Music on hold is stoped.
+			/// </summary>
+			Stop
+		}
+
+		/// <summary>
+		/// Get or set state
+		/// </summary>
+		public MusicOnHoldStates State { get; set; }
```
[the remark this answers is not in the document]
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs]: I have not tested. It was a direct copy of the earlier solution. I can work on verifying

File: Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿namespace AsterNET.Manager.Event
+{
+	/// <summary>
+	/// The MusicOnHoldEvent event triggers when the music starts or ends playing the hold music.
```
[the remark this answers is not in the document]
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs]: I can check if i have made the change in the other commit

File: Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs [hunk 3]
```diff
@@ -1314,13 +1319,13 @@ public Response.ManagerResponse SendAction(Action.ManagerAction action)
         /// <param name="action">action to send</param>
         /// <param name="timeout">timeout in milliseconds</param>
         /// <returns></returns>
-        public Response.ManagerResponse SendAction(ManagerAction action, int timeOut)
+        public Response.ManagerResponse SendAction(ManagerAction action, int timeout)
```
[the remark this answers is not in the document]
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs]: will do. follows the rest of the timeout convention (ie. all the others are not camel case)

File: Asterisk.2013/Documentation/Documentation.shfbproj [hunk 4]
```diff
@@ -7,7 +7,7 @@
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <SchemaVersion>2.0</SchemaVersion>
     <ProjectGuid>186124f7-82fa-4562-8b87-33d53a84e8ba</ProjectGuid>
-    <SHFBSchemaVersion>1.9.9.0</SHFBSchemaVersion>
+    <SHFBSchemaVersion>2017.9.26.0</SHFBSchemaVersion>
```
[the remark this answers is not in the document]
Comment (moldypenguins) [reply Asterisk.2013/Documentation/Documentation.shfbproj]: VS did this one, can be disregarded really

[the remark this answers is not in the document]
Comment (moldypenguins) [reply Asterisk.2013/Documentation/Documentation.shfbproj]: doc_upgrade


Documentation\Documentation.shfbproj: The project file was upgraded successfully.

**Comment by moldypenguins** [comment]
The MusicOnHold event is now two events: MusicOnHoldStart and MusicOnHoldStop. The sub type field has been removed.

I will have to review the whole commit. it seems things were left out.

**Comment by moldypenguins** [comment]
BridgeEvent currently has an enum in the class. I think maybe there needs to be a bit of clarification regarding enums in event classes. (ie. if they don't work they should be removed, if they do work or plan to be working in the future then leave them in)

**Comment by Deantwo** [comment]
I don't think enums are used in any of the other/current event or command classes.

**Comment by moldypenguins** [comment]
See Manager > Event > BridgeEvent.cs
```
        public enum BridgeStates
		{
			Unknown,
			BRIDGE_STATE_LINK,
			BRIDGE_STATE_UNLINK
		}
```

**Comment by moldypenguins** [comment]
Well as long as we know it does work, we can move forward a bit

**Comment by Deantwo** [comment]
Don't know if it will work.
I would still like to hear from someone who has tested this.

File: Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs [hunk 5]
```diff
@@ -1331,6 +1336,12 @@ public Response.ManagerResponse SendAction(ManagerAction action, int timeOut)
         #endregion
 
         #region SendAction(action, responseHandler)
+        /// <summary>
+        /// Send action ans with timeout (milliseconds)
```
Comment (Deantwo) [inline Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs]: Typo?

File: Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+﻿namespace AsterNET.Manager.Event
+{
+    /// <summary>
+    ///     The MusicOnHoldEvent event triggers when the music starts or ends playing the hold music.<br />
+    ///     See <see target="_blank"  href="LINK">LINK</see>
```
Comment (Deantwo) [inline Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs]: Why is this link blank?

File: Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs (hunk 5 above)
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs]: auto comment (ans = answer) 
i'll fix

File: Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs (hunk 6 above)
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs]: forgot to fill it in
will fix

File: Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs (hunk 5 above)
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs]: took this out, will add it to another pull request
also took out the timeOut rename, will do that in another request too

File: Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs (hunk 6 above)
Comment (moldypenguins) [reply Asterisk.2013/Asterisk.NET/Manager/Event/MusicOnHoldEvent.cs]: I can't really find docs for that event
I removed it for start/stop events

**Comment by moldypenguins** [comment]
should be okay now

**Pull request merged**